include/revng/Model/PrimitiveType.h

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Model/Register.h"
+#include "revng/Model/Type.h"
+
+/* TUPLE-TREE-YAML
+name: PrimitiveType
+type: struct
+inherits: Type
+fields:
+  - name: PrimitiveKind
+    type: PrimitiveKind
+  - name: Size
+    type: uint64_t
+TUPLE-TREE-YAML */
+
+#include "revng/Model/Generated/Early/PrimitiveType.h"
+
+class model::PrimitiveType : public model::generated::PrimitiveType {
+public:
+  static constexpr const auto AssociatedKind = TypeKind::PrimitiveType;
+
+public:
+  using generated::PrimitiveType::PrimitiveType;
+
+  //
+  // General construction
+  //
```

Comment:
When you want to group methods, sometimes we use [Doxygen grouping](https://github.com/revng/revng/blob/develop/include/revng/Support/MetaAddress.h#L448).